content/articles/creating-chatbot-using-natural-language-processing-in-python/index.md

```diff
@@ -0,0 +1,239 @@
+A bot is a computer program that performs predetermined tasks automatically. It is the goal of the bots to perform human duties in the same way that humans did in the past. In a nutshell, they mimic human behaviour.
+
+We'll design a conversational interface for our chatbot using natural language processing in this tutorial.
+
+### Prerequisites
+- Have a basic knowledge of [Python](https://www.python.org/).
+- Have python environment of your choice installed.
+
+### Table of contents
+- [Fields of NLP](#fields-of-nlp)
+- [Types of bots](#types-of-bots)
+- [Building an NLP chatbot](#building-an-nlp-chatbot)
+  - [Importing modules](#importing-modules)
+  - [Creating jason file](#creating-jason-file)
+  - [Processing data](#processing-data)
+  - [Designing a Neural Network model](#designing-a-neural-network-model)
+  - [Building useful features](#building-useful-features)
+- [Benefits of bots](#benefits-of-bots)  
+- [Conclusion](#conclusion)
+- [Reference](#reference)
+
+### Fields of NLP
+Below are the core fields of NLP:
+
+1. **Natural language generation(NLG)**
+
+In artificial intelligence, NLG is a speciality. It's a software technology that automatically transforms data into simple English. 
+
+2. **Natural language understanding(NLU)**
+
+It is a branch of natural language processing (NLP) that helps computers understand and interpret human language by breaking speech into its constituent parts. However, NLU goes further than speech recognition to understand what the user is trying to communicate with their words.
+
+3. **Natural language interaction(NLI)**
+
+To engage with any connected device or service in a humanlike manner, NLI brings together a varied collection of natural language principles.
+
+### Types of bots
+- **Scraper Bots:** They are designed to read and save data from the Internet.
+- **Spam Bots:** Designed to gather email addresses from spam mailing lists.
+- **Spider Bots:** Searches the web and downloads pages before indexing the material on each page it comes across.
+- **Transnational Bots:** Designed to be used in transactions.
+- **Social Media Bot:** Automatically responds to questions on social media platforms.
+
+### Building an NLP chatbot
+The first step is to create an intents JSON file that lists all the possible outcomes of user interactions with our chatbot. We first need a set of tags that users can use to categorize their queries. These tags include `name`, `age` etc. Every new tag would require its own unique pattern. Identifying these trends can help the chatbot train itself on how people query about our chatbot's name, allowing it to be more responsive. To answer questions, the chatbot will return pre-programmed responses.
+
+#### Importing modules

```

Comment:
Clearly highlight your steps. I assume this is step one. Do the same for the rest.